src/post.ts

```diff
@@ -0,0 +1,111 @@
+import * as https from "https";
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+
+const sendReportRetryCount: number = 1;
+
+export async function run() {
+    let startTime = core.getState('PreJobStartTime');
+    if (startTime.length <= 0) {
+        console.log(`PreJobStartTime not defined, using now-10secs `);
+        startTime = new Date(new Date().getTime() - 10000).toISOString();
+    }
+
+    let dockerVer = Buffer.alloc(0);
+    let dockerEvents = Buffer.alloc(0);
+    let dockerImages = Buffer.alloc(0);
+    
+    // Initialize the commands 
+    await exec.exec('docker --version', null, {
+        listeners: {
+            stdout: (data: Buffer) => {
+                dockerVer = Buffer.concat([dockerVer, data]);
+            }
+        }
+    });
+    await exec.exec(`docker events --since ${startTime} --until ${new Date().toISOString()} --filter event=push --filter type=image --format ID={{.ID}}`, null, {
+        listeners: {
+            stdout: (data: Buffer) => {
+                dockerEvents = Buffer.concat([dockerEvents, data]);
+            }
+        }
+    });
+    await exec.exec('docker images --format CreatedAt={{.CreatedAt}}::Repo={{.Repository}}::Tag={{.Tag}}::Digest={{.Digest}}', null, {
+        listeners: {
+            stdout: (data: Buffer) => {
+                dockerImages = Buffer.concat([dockerImages, data]);
+            }
+        }
+    });
+
+    // Post data to URI
+    let data = {
+        dockerVer: dockerVer.toString(),
+        dokcerEvents: dockerEvents.toString(),

```

Comment:
the backend expects an array of lines rather than a big string. It would be great if we can we send it like that.